gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -630,21 +630,38 @@ public void rename(URI src, URI dst) throws IOException {
           StorageResourceId.fromUriPath(dst, /* allowEmptyObjectName= */ true);
 
       gcs.copy(
-          srcResourceId.getBucketName(), ImmutableList.of(srcResourceId.getObjectName()),
-          dstResourceId.getBucketName(), ImmutableList.of(dstResourceId.getObjectName()));
-
-      // TODO(b/110833109): populate generation ID in StorageResourceId when getting info
-      gcs.deleteObjects(
+          ImmutableList.of(srcResourceId),
           ImmutableList.of(
               new StorageResourceId(
-                  srcInfo.getItemInfo().getBucketName(),
-                  srcInfo.getItemInfo().getObjectName(),
-                  srcInfo.getItemInfo().getContentGeneration())));
+                  dstResourceId.getBucketName(),
```

Comment:
What `generationId` has `generationId` object? Can we set it to desired value when we create this object upstream instead?

Reply:
We only receive `URI` from upstream, setting `generationId=0` would ensure that the dst file does not exist.

Could you elaborate? may be i don't follow the comment

Reply:
Is `generationId=0` a default value for `StorageResourceId` objects?

Reply:
bump